sunpy/map/maphelper.py

```diff
@@ -0,0 +1,253 @@
+            else:
+                print(k, " is not a meta keyword required for sunpy.map.Map \n The map meta keywords are : \n", list(map_meta_keywords.keys()))
+        return
+
+    for k in map_meta_keywords:
+        print(k, " : ", map_meta_keywords[k])
+
+
+@u.quantity_input
+def make_fitswcs_header(data, coordinate, reference_pixel: u.pix = None, scale: u.arcsec/u.pix = None, **kwargs):
+    """
+    Function to create a `sunpy.util.meta.MetaDict` from a coordinate object (`astropy.coordinates.SkyCoord`
+    or a coordinate frame) that is required to create a `sunpy.map.Map`
+
+    Parameters
+    ----------
+    data : `~numpy.ndarray`
+        Array data of Map for which a header is required.
+    coordinates : `~astropy.coordinates.SkyCoord` or `~astropy.coordinates.BaseFrame`
+        Coordinate object to get meta information for map header.
+    reference_pixel :`~astropy.units.Quantity` of size 2, optional
+        Reference pixel along each axis. These are expected to be Cartestian ordered, i.e
+        the first index is the x axis, second index is the y axis.
+        Defaults to the center of data array, ``(data.shape[1] + 1)/2., (data.shape[0] + 1)/2.)``.
+    scale : `~astropy.units.Quantity` of size 2, optional
+        Pixel scaling along x and y axis (i.e. the spatial scale of the pixels (dx, dy)). These are
+        expected to be Cartestian ordered, i.e [dx, dy].
+        Defaults to ``([1., 1.] arcsec/pixel)``.
+    **kwargs:
+        Additional arguments that will be put into the metadict header. These must be in the
+        list of map meta keywords. The keyword arguments for the instrument meta can also be given 
+        in certain forms which will be translated to fits standard:
+            * ``instrument`` 
+            * ``telescope``
+            * ``observatory``
+            * ``wavelength``
+            * ``exposure``
+
+    Returns
+    -------
+    `~sunpy.util.MetaDict`
+        A `sunpy.util.MetaDict` that contains the header information required for making a `sunpy.map.Map`
+
+    Examples
+    --------
+    >>> import sunpy.map
+    >>> from sunpy.coordinates import frames
+    >>> from astropy.coordinates import SkyCoord
+    >>> from astropy import units as u
+    >>> import numpy as np
+
+    >>> data = np.random.rand(1024, 1024)
+    >>> my_coord = SkyCoord(0*u.arcsec, 0*u.arcsec, obstime="2017-08-01", observer = 'earth', frame=frames.Helioprojective)
+    >>> my_header = sunpy.map.make_fitswcs_header(data, my_coord)
+    >>> my_map = sunpy.map.Map(data, my_header)
+    """
+
+    if not isinstance(coordinate, (SkyCoord, frames.BaseCoordinateFrame)):
+        raise ValueError("coordinate needs to be a coordinate frame or an SkyCoord instance.")
+
+    if coordinate.obstime is None:
+        raise ValueError("The coordinate needs an observation time, `obstime`.")
+
+    if isinstance(coordinate, SkyCoord):
+        coordinate = coordinate.frame
+
+    if coordinate.name == 'heliocentric':
```

Comment:
I would you replace this with an `isinstance` check: (I think that's the correct frame name)
```suggestion
    if isinstance(coordinate, frames.Heliocentric):
```